tools/ab_test.py

```diff
@@ -318,20 +314,21 @@ def ab_performance_test(
     )
     print(commit_list.strip())
 
-    git_ab_test_with_binaries(
-        lambda firecracker_binary, jailer_binary: collect_data(
-            firecracker_binary, jailer_binary, test
-        ),
-        lambda ah, be: analyze_data(
-            ah,
-            be,
-            p_thresh,
-            strength_abs_thresh,
-            noise_threshold,
-            n_resamples=int(100 / p_thresh),
-        ),
-        a_revision=a_revision,
-        b_revision=b_revision,
+    test = ".." / Path(test)
+    res = []
+    for rev in [a_revision, b_revision]:
+        print(f"Checking out and building {rev}")
+        bin_dir = git_clone(Path("build") / rev, rev)
+        res.append(collect_data(bin_dir, test))
+
+    processed_emf_a, processed_emf_b = res[0], res[1]
+    return analyze_data(
+        processed_emf_a,
+        processed_emf_b,
+        p_thresh,
+        strength_abs_thresh,
+        noise_threshold,
+        n_resamples=int(100 / p_thresh),
     )
```

Comment:
```suggestion
    git_ab_test(
        lambda workspace, _: collect_data(
            get_binary("firecracker", workspace_dir=workspace),
            get_binary("jailer", workspace_dir=workspace),
        ),
        lambda ah, be: analyze_data(
            ah,
            be,
            p_thresh,
            strength_abs_thresh,
            noise_threshold,
            n_resamples=int(100 / p_thresh),
        ),
    )
```

Reply:
I think the 1st version is a bit more readable, but let's go with yours so we centralize ab_test in one place.